src/components/MessageList/MessageList.js

```diff
@@ -100,19 +72,38 @@ const MessageList = (props) => {
     TypingIndicator = DefaultTypingIndicator,
   } = props;
 
-  const messages = threadList ? threadMessages : mainMessages;
-  const loadMore = threadList ? mainLoadMore : loadMoreThread;
-  const read = threadList ? {} : mainRead;
+  const { t } = useContext(TranslationContext);
+  const { client, isOnline } = useContext(ChatContext);
+  const {
+    Attachment,
+    clearEditingState,
+    editing,
+    emojiData,
+    loadMore: mainLoadMore,
+    Message,
+    removeMessage,
+    retrySendMessage,
+    setEditingState,
+    updateMessage,
+  } = useContext(MessagesContext);
+  const { loadMoreThread, openThread } = useContext(ThreadContext);
+  const { channel, disabled, EmptyStateIndicator, markRead } = useContext(
+    ChannelContext,
+  );
+  const messageList = useMessageList(threadList, noGroupByUser);
```

Comment:
eventually I'd like for us to be using `useMessageList({ noGroupByUser, threadList });` as a single object param instead of the multiple ones so naming is consistent across files and ordering is less of an issue that we need to remember. but for a future change

Reply:
that's new so did it now